lib/PhasarLLVM/TypeHierarchy/DIBasedTypeHierarchy.cpp

```diff
@@ -189,6 +219,84 @@ DIBasedTypeHierarchy::DIBasedTypeHierarchy(const LLVMProjectIRDB &IRDB) {
   buildTypeHierarchy(TG, VertexTypes, TransitiveDerivedIndex, Hierarchy);
 }
 
+static const llvm::DICompositeType *
+stringToDICompositeType(const llvm::DebugInfoFinder &DIF,
+                        llvm::StringRef DITypeName) {
+  const llvm::DICompositeType *ByName = nullptr;
+  bool HasUniqueByName = false;
+  for (const auto *Type : DIF.types()) {
+    if (const auto *DICT = llvm::dyn_cast<llvm::DICompositeType>(Type)) {
+      auto Ident = DICT->getIdentifier();
+      if (Ident == DITypeName) {
+        return DICT;
+      }
+      if (DICT->getName() == DITypeName) {
+        HasUniqueByName = ByName == nullptr;
+        ByName = DICT;
+      }
+    }
+  }
+  if (HasUniqueByName) {
+    return ByName;
+  }
+
+  llvm::report_fatal_error("DIType doesn't exist: " + DITypeName);
+}
+
+static const llvm::DIType *stringToDIType(const llvm::DebugInfoFinder &DIF,
+                                          llvm::StringRef DITypeName) {
+  for (const auto *Type : DIF.types()) {
+    if (Type->getName() == DITypeName) {
+      return Type;
+    }
+  }
+
+  llvm::report_fatal_error("DIType doesn't exist " + DITypeName);
+}
+
+// NOLINTNEXTLINE
+static constexpr char NullFunName[] = "__null__";
```

Comment:
this is the second time I see this definition. Please use a single definition and make it static inline. also, again you can go for auto and avoid the c-style array.

Reply:
fixed